core/src/main/java/org/apache/carbondata/core/scan/collector/impl/DictionaryBasedVectorResultCollector.java

```diff
@@ -51,8 +51,8 @@
 
   public DictionaryBasedVectorResultCollector(BlockExecutionInfo blockExecutionInfos) {
     super(blockExecutionInfos);
-    queryDimensions = tableBlockExecutionInfos.getQueryDimensions();
-    queryMeasures = tableBlockExecutionInfos.getQueryMeasures();
+    queryDimensions = tableBlockExecutionInfos.getActualQueryDimensions();
```

Comment:
getQueryDimensions or getQueryMeasures does not contain the newly added columns so when we try to get the vector from allColumnInfo which has the size [measures + dimensions] we would get a ArrayIndexOutOfBoundException because it will not have the space for the new column.